internal/resources/helmfeature/resource_mock_test.go

```diff
@@ -0,0 +1,328 @@
+		if err = json.Unmarshal(bodyBytes, &bodyInterface); err == nil {
+			var expectedInterface map[string]interface{}
+
+			err = json.Unmarshal(expectedBytes, &expectedInterface)
+			if err != nil {
+				return nil, err
+			}
+
+			diff := deep.Equal(bodyInterface, expectedInterface)
+			if diff == nil {
+				return successFunc()
+			}
+		} else {
+			return nil, err
+		}
+
+		return successFunc()
+	}
+}
+
+// Register a new responder when the given call is made.
+func changeStateResponder(registerFunc func(), successResponse int, successResponseBody interface{}) httpmock.Responder {
+	return func(r *http.Request) (*http.Response, error) {
+		registerFunc()
+		return httpmock.NewJsonResponse(successResponse, successResponseBody)
+	}
+}
+
+func (testConfig *testAcceptanceConfig) setupHTTPMocks(t *testing.T) {
+	httpmock.Activate()
+	t.Cleanup(httpmock.Deactivate)
+
+	endpoint := os.Getenv("TMC_ENDPOINT")
+
+	OrgID := os.Getenv("ORG_ID")
+
+	reference := objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference{
+		Rid: "test_rid",
+		UID: "test_uid",
+	}
+	referenceArray := make([]*objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference, 0)
+	referenceArray = append(referenceArray, &reference)
+
+	// cluster level Helm resource.
+	postHelmFeatureRequest, postHelmFeatureResponse, listResponse := testConfig.getClRequestResponse(OrgID, referenceArray)
+
+	listDeleteResp := &helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmListHelmsResponse{
+		Helms: []*helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmHelm{
+			{
+				FullName: &helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmFullName{
+					ClusterName:           testConfig.ScopeHelperResources.Cluster.Name,
+					OrgID:                 OrgID,
+					ManagementClusterName: "attached",
+					ProvisionerName:       "attached",
+				},
+				Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+					ParentReferences: referenceArray,
+					Description:      "resource with description",
+					Labels: map[string]string{
+						"key1": "value1",
+						"key2": "value2",
+					},
+					UID:             "gitrepository1",
+					ResourceVersion: "v1",
+				},
+				Status: &helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmStatus{
+					Conditions: map[string]statusmodel.VmwareTanzuCoreV1alpha1StatusCondition{
+						"Disabled": {
+							Reason: "feature is disable on the cluster",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	HelmFeatureEndpoint := (helper.ConstructRequestURL(https, endpoint, clAPIVersionAndGroup, testConfig.ScopeHelperResources.Cluster.Name, helmAPIKind)).String()
+
+	httpmock.RegisterResponder("POST", HelmFeatureEndpoint,
+		bodyInspectingResponder(t, postHelmFeatureRequest, 200, postHelmFeatureResponse))
+
+	httpmock.RegisterResponder("GET", HelmFeatureEndpoint,
+		bodyInspectingResponder(t, nil, 200, listResponse))
+
+	// httpmock.RegisterResponder("DELETE", HelmFeatureEndpoint,
+	// 	bodyInspectingResponder(t, nil, 200, nil))
```

Comment:
Can you remove the commented blocks if not required?

Reply:
Updated PTAL